src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Logger, SfdxError, LoggerLevel} from '@salesforce/core';
+import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
+import {OutputProcessor} from '../pmd/OutputProcessor';
+import {RuleEngine} from '../services/RuleEngine';
+import {CLIEngine} from 'eslint';
+import * as path from 'path';
+
+
+export interface EslintStrategy {
+
+	init(): Promise<void>;
+
+	getName(): string;
+
+	isEnabled(): boolean;
+
+	getTargetPatterns(target?: string): Promise<string[]>;
+
+	getCatalogConfig(): Record<string, any>;
+
+	getRunConfig(target?: string): Promise<Record<string, any>>;
+
+	getLanguage(): string[];
+
+	isRuleKeySupported(key: string): boolean;
+
+	filterUnsupportedPaths(paths: string[]): string[];
+}
+export class StaticDependencies {
```

Comment:
A first reading of this file leaves me confused why the `StaticDependencies` class has no static methods. Why is that?

Reply:
The methods in StaticDependencies encapsulate a static/irreplaceable piece of code. While testing, we can mock the methods in StaticDependency to control how the test works. 
For example, process.cwd() would always return the current working directory. But we may want this to return a different value in our test without having to change the structure of the current working directory for real to suit the test. Now that it has been pulled away as a StaticDependency, our test can mock the value that'll be returned as current directory. Hope that makes sense?
This is a fairly common testing technique and the naming convention typically called the pull-away class as "StaticDependencies". So I'm just sticking with the norm.